apps/idos-enclave/src/lib/enclave.js

```diff
@@ -176,7 +178,7 @@ export class Enclave {
   }
 
   async decrypt(fullMessage, senderPublicKey) {
-    if (!this.keyPair) await this.keys();
+    if (!this.keyPair) await this.keys(senderPublicKey);
```

Comment:
This is not true. :x You're telling the code the user should have the same pub key as the encrypter, which is false.